.github/workflows/pathogen-repo-ci.yaml

```diff
@@ -22,10 +22,42 @@ on:
         default: ${{ github.repository }}
         required: false
 
+      env:
+        description: >-
+          Additional environment variables to set before the build, as a string
+          containing YAML.  This is easily produced, for example, by pretending
+          you're writing normal nested YAML within a literal multi-line block
+          scalar (introduced by "|"):
+
+            with:
+              env: |
+                FOO: bar
+                I_CANT_BELIEVE: "it's not YAML"
+
+          Do not use for secrets!  Instead, pass them via GitHub Action's
+          dedicated secrets mechanism.
+        type: string
+        default: ""
+        required: false
+
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
+      - if: inputs.env
+        name: Set environment variables
+        env:
+          env: ${{ inputs.env }}
+        run: >
+          echo "$env"
+          | yq --output-format json .
+          | jq --raw-output '
+              to_entries
+            | map("\(.key)<<__EOF__\n\(.value)\n__EOF__")
```

Comment:
_[non-blocking]_

Is this heredoc for variables? How'd you know it works for `$GITHUB_ENV`? I tried running it directly:

```sh
I_CANT_BELIEVE<<__EOF__
it's not YAML
__EOF__
# bash: I_CANT_BELIEVE: command not found
```

Reply:
It's not standard Bash, but something that GitHub Actions implements, see https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings.

Reply:
Added a comment about this in [c90c1b1](https://github.com/nextstrain/.github/pull/23/commits/c90c1b18f47ac69c9a9ba115f4f1e5083beda810).